### PR TITLE
Add migration for IS590

### PR DESCRIPTION
**Pull request by janakamarasena**

### CODE REVIEW

File: components/org.wso2.is.migration/migration-resources/5.9.0/data/claim-config.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  ~
+  -->
+
+<ClaimConfig>
+    <Dialects>
+        <Dialect dialectURI="http://wso2.org/claims">
+            <Claim>
```
Comment (tharindu-bandara) [inline components/org.wso2.is.migration/migration-resources/5.9.0/data/claim-config.xml]: Shall we add a comment to describe this claim as well?

File: components/org.wso2.is.migration/migration-resources/5.9.0/dbscripts/step2/carbon/oracle.sql [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+/
+ALTER TABLE UM_SYSTEM_ROLE ADD CONSTRAINT UC_UM_SYSTEM_ROLE UNIQUE(UM_ROLE_NAME,UM_TENANT_ID)
+/
+
+ALTER TABLE UM_SYSTEM_ROLE DROP CONSTRAINT UNIQUE_REG_PATH_TENANT_ID
+/
+ALTER TABLE UM_SYSTEM_ROLE ADD CONSTRAINT UNIQUE_REG_PATH_TENANT_ID UNIQUE(REG_PATH_VALUE,REG_TENANT_ID)
+/
+
+DROP INDEX UM_ROLE_NAME_IND
```
Comment (tharindu-bandara) [inline components/org.wso2.is.migration/migration-resources/5.9.0/dbscripts/step2/carbon/oracle.sql]: Does the existence of the index is guaranteed? Better to check if Oracle throws up an error if not

File: components/org.wso2.is.migration/migration-resources/migration-config.yaml [hunk 3]
```diff
@@ -298,3 +298,20 @@ versions:
...
+     name: "SchemaMigrator"
+     order: 1
+     parameters:
+       location: "step1"
+       schema: "identity"
+   -
+     name: "SchemaMigrator"
+     order: 2
+     parameters:
+       schema: "identity"
```
Comment (tharindu-bandara) [inline components/org.wso2.is.migration/migration-resources/migration-config.yaml]: Location parameter?

File: components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v590/migrator/ClaimDataMigrator.java [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+                mappedAttributes.add(new AttributeMapping(claimMappingEntry.getKey(), claimMappingEntry.getValue()));
+            }
+        }
+
+        Map<String, String> claimProperties = claimConfig.getPropertyHolder().get(claimURI);
+        claimProperties.remove(ClaimConstants.DIALECT_PROPERTY);
+        claimProperties.remove(ClaimConstants.CLAIM_URI_PROPERTY);
+        claimProperties.remove(ClaimConstants.ATTRIBUTE_ID_PROPERTY);
+
+        if (!claimProperties.containsKey(ClaimConstants.DISPLAY_NAME_PROPERTY)) {
```
Comment (tharindu-bandara) [inline components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v590/migrator/ClaimDataMigrator.java]: Are the claim properties read from the provided claim-config.xml file? If so are we overriding the display name property value set from the file?

Comment (janakamarasena) [reply components/org.wso2.is.migration/migration-service/src/main/java/org/wso2/carbon/is/migration/service/v590/migrator/ClaimDataMigrator.java]: Yes, it's taken from the file... its only set if the display name is not available.

File: components/org.wso2.is.migration/migration-resources/migration-config.yaml (hunk 3 above)
Comment (janakamarasena) [reply components/org.wso2.is.migration/migration-resources/migration-config.yaml]: fixed

File: components/org.wso2.is.migration/migration-resources/5.9.0/dbscripts/step2/carbon/oracle.sql (hunk 2 above)
Comment (janakamarasena) [reply components/org.wso2.is.migration/migration-resources/5.9.0/dbscripts/step2/carbon/oracle.sql]: Updated the queries

File: components/org.wso2.is.migration/migration-resources/5.9.0/data/claim-config.xml (hunk 1 above)
Comment (janakamarasena) [reply components/org.wso2.is.migration/migration-resources/5.9.0/data/claim-config.xml]: Fixed

File: components/org.wso2.is.migration/migration-resources/5.9.0/dbscripts/step1/identity/oracle.sql [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+      PRIMARY KEY (TENANT_ID,NAME))
+/
+
+CREATE OR REPLACE PROCEDURE add_index_if_not_exists (query IN VARCHAR2)
+  IS
+BEGIN
+  execute immediate query;
+  dbms_output.put_line(query);
+exception WHEN OTHERS THEN
+  dbms_output.put_line( 'skipped' );
```
Comment (tharindu-bandara) [inline components/org.wso2.is.migration/migration-resources/5.9.0/dbscripts/step1/identity/oracle.sql]: Is it possible to add some context about the scenario, to the dbms_output when the exception occurs?

Comment (janakamarasena) [reply components/org.wso2.is.migration/migration-resources/5.9.0/dbscripts/step1/identity/oracle.sql]: Added

File: components/org.wso2.is.migration/migration-resources/5.9.0/dbscripts/step1/identity/oracle.sql [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      PRIMARY KEY (TENANT_ID,NAME))
+/
+
+CREATE OR REPLACE PROCEDURE add_index_if_not_exists (query IN VARCHAR2, t_index IN VARCHAR2)
+  IS
+BEGIN
+  execute immediate query;
+  dbms_output.put_line(query);
+exception WHEN OTHERS THEN
+  dbms_output.put_line( 'skipped '|| t_index ||' index creation as index is already available' );
```
Comment (tharindu-bandara) [inline components/org.wso2.is.migration/migration-resources/5.9.0/dbscripts/step1/identity/oracle.sql]: the exception might occur due to some reason other than the available index, IMO. If possible, can we add error code into this? A similar example is found in https://docs.oracle.com/cd/B19306_01/appdev.102/b14261/sqlcode_function.htm.
```
DECLARE
   name employees.last_name%TYPE;
   v_code NUMBER;
   v_errm VARCHAR2(64);
BEGIN
   SELECT last_name INTO name FROM employees WHERE employee_id = 1000;
   EXCEPTION
      WHEN OTHERS THEN
         v_code := SQLCODE;
         v_errm := SUBSTR(SQLERRM, 1 , 64);
         DBMS_OUTPUT.PUT_LINE('The error code is ' || v_code || '- ' || v_errm);
END;
/
```

Comment (janakamarasena) [reply components/org.wso2.is.migration/migration-resources/5.9.0/dbscripts/step1/identity/oracle.sql]: Added

**Pull request merged**